File: src/api/routes/collections.py
from fastapi import APIRouter, Request, Depends, Query
from typing import Annotated, List

from src.api.deps import RepoDep
from src.api.schemas import AddCollectionSchema, CollectionSchema

router = APIRouter(prefix="/api/v1/collections", tags=["Vector Repository"])


@router.post("/", summary="Создать коллекцию")
async def create_collection(collection: AddCollectionSchema, repo: RepoDep):
    await repo.create_collection(collection.name, collection.size, collection.distance)
    return {"message": f"Коллекция '{collection.name}' создана успешно"}


@router.get("/", summary="Получить список коллекций")
async def get_collections(
    repo: RepoDep, 
    include_parents: bool = Query(
        default=False, 
        description="Включать ли служебные родительские коллекции (*_parents)"
    )
) -> List[CollectionSchema]:
    return await repo.get_collections(include_parents=include_parents)


@router.get("/{collection_name}", summary="Получить данные о коллекции")
async def get_collection_details(collection_name: str, repo: RepoDep):
    return await repo.get_collection_details(collection_name)


@router.delete("/{collection_name}/points", summary="Очистить коллекцию")
async def clear_collection(collection_name: str, repo: RepoDep):
    await repo.clear_collection(collection_name)


@router.delete("/{collection_name}", summary="Удалить коллекцию")
async def delete_collection(collection_name: str, repo: RepoDep):
    await repo.delete_collection(collection_name)

File: frontend/src/api/collections.ts
import client from "./client";
import type { CollectionSummary, CollectionDetails } from "@/types/collection";

export interface CreateCollectionPayload {
  name: string;
  size?: number;
  distance?: string;
  description?: string;
}

export interface UpdateCollectionPayload {
  name?: string;
  description?: string;
}


export type CollectionId = string;

export const fetchCollections = async (): Promise<CollectionSummary[]> => {
  const { data } = await client.get<CollectionSummary[]>("/collections");
  return data;
};

export const fetchCollection = async (collectionId: CollectionId): Promise<CollectionDetails> => {
  const { data } = await client.get<CollectionDetails>(`/collections/${collectionId}`);
  return data;
};

export const createCollection = async (payload: CreateCollectionPayload): Promise<CollectionDetails> => {
  const { data } = await client.post<CollectionDetails>("/collections", payload);
  return data;
};

export const updateCollection = async (collectionId: CollectionId, payload: UpdateCollectionPayload): Promise<CollectionDetails> => {
  const { data } = await client.patch(`/collections/${collectionId}`, payload);
  return data;
};

export const deleteCollection = async (collectionId: CollectionId): Promise<void> => {
  await client.delete(`/collections/${collectionId}`);
};

export const clearCollection = async (collectionId: CollectionId): Promise<void> => {
  await client.delete(`/collections/${collectionId}/points`);
};
